### PR TITLE
slightly improve registerCallback API, so it returns a full Result

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Improve internal API for registerCallback, so it returns a full Result with an error message in case of error. In addition, add the `[[nodiscard]]` attribute to the API and make sure its return value is checked everywhere.

- [x] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [x] No backports required

### Testing & Verification

- [x] The behavior in this PR was *manually tested*
- [x] This change is already covered by existing tests, such as *everything in cluster*.

Link to Jenkins PR run:
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13629/

### CODE REVIEW

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13630/

**Review by neunhoef — APPROVED** [review]
LGTM

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13636/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13639/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13642/

File: arangod/Cluster/AgencyCache.cpp [hunk 1]
```diff
@@ -512,7 +512,7 @@ bool AgencyCache::registerCallback(std::string const& key, uint64_t const& id) {
   LOG_TOPIC("31415", TRACE, Logger::CLUSTER)
     << "Registered callback for key " << ckey << " with id " << id << ", callbacks: " << size;
   // this method always returns ok.
-  return true;
+  return {};
 }
```
Comment (goedderz) [inline arangod/Cluster/AgencyCache.cpp:516]: Does it make sense to keep a return value at all, if this never returns an error?

Comment (jsteemann) [reply arangod/Cluster/AgencyCache.cpp]: We could adjust the return value of this method to `void`, but I would like to return a `Result` here, so we can keep return type compatibility between AgencyCache::registerCallback(...) and AgencyCallbackRegistry::registerCallback(...).

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13651/

**Review by neunhoef — APPROVED** [review]
LGTM

**Pull request merged**